### PR TITLE
CoAP support

### PR DESCRIPTION
**Pull request by arturkow2000**

### CODE REVIEW

File: pal/pal_nrf/src/lib.rs [hunk 1]
```diff
@@ -81,7 +84,10 @@ pub fn init() {
     );
 
     // configure TIMER2 to be used for delays
```
Comment (krystian-hebel) [inline pal/pal_nrf/src/lib.rs:86]: And what is TIMER3 used for?

File: src/main.rs [hunk 2]
```diff
@@ -10,18 +10,66 @@ extern crate pal_pc as pal;
...
+const UDP_META_DEFAULT_BUF_LEN: usize = 16;
+const ECHO_SERVER_BUF_LEN: usize = 512;
+// Set at Ethernet MTU
+// Actually we can transfer less than that in UDP packet since Ethernet frame +
+// IP frame already takes some space. Should compute exatly how much bytes do we
+// need.
+const COAP_CLIENT_BUF_LEN: usize = 1500;
+
+// FIXME: should not use hardcoded server addresses, instead server should be
+// discovered
```
Comment (krystian-hebel) [inline src/main.rs]: Actually, Fobnail should make host use this (or any other in this network) address so it won't have to be manually assigned by user. Maybe hardcoded response to DHCP query from host?

**Review by krystian-hebel — APPROVED** [review]
I haven't looked too deep into the code but compiled application seems to work so I'm giving weak approve.

File: pal/pal_nrf/src/lib.rs (hunk 1 above)
Comment (arturkow2000) [reply pal/pal_nrf/src/lib.rs:86]: I updated comment:
```
// configure TIMER2 to be used for delays
// configure TIMER3 as a freerunning monotonic counter
```

File: src/main.rs (hunk 2 above)
Comment (arturkow2000) [reply src/main.rs]: @krystian-hebel
> Actually, Fobnail should make host use this (or any other in this network) address so it won't have to be manually assigned by user.

Right.

> Maybe hardcoded response to DHCP query from host?

I'm not sure whether that's possible, IIRC response contains sender's MAC address, and maybe something more.
Some time ago I implemented DHCP server in my personal project https://github.com/arturkow2000/pxeserver/blob/main/src/dhcp/mod.rs which may serve as a reference.

Anyway, I updated the comment: 
```
// FIXME: server should have IP address assigned by using DHCP server located in
// Fobnail token
```

Comment (krystian-hebel) [reply src/main.rs]: Right, by hardcoded I meant just IP, without having to implement a database, assignment algorithm etc. on DHCP server's side.

Ack on the updated comment.

**Pull request merged**